sources/api/shibaken/src/main.rs

```diff
@@ -191,7 +158,7 @@ fn parse_args(args: env::Args) -> Result<Args> {
     }
 
     Ok(Args {
-        log_level: log_level.unwrap_or_else(|| LevelFilter::Info),
+        log_level: log_level.unwrap_or(LevelFilter::Info),
```

Comment:
Since `sundog` only prints `stderr` in the event of an error, would it make sense for me to change default log level to Debug? It would make it easier to troubleshoot, but might be somewhat ugly if someone ran `shibaken` in a shell.

Reply:
It's kind of an edge-case, but the current IMDS logic in Bottlerocket treats a 404 as an `Ok(None)` and that's what we want 99% of the time. However, in `shibaken` we decided to actually error out if it receives a 404 when it is collecting the data from the individual keys in IMDS. Since the `imdsclient` isn't returning an error on 404, some context is lost.

At an error level `shibaken` would return something like:
```
Error retrieving 'public-keys/1/openssh-key' from IMDS metadata
```
but the debug message would show the full uri like this:
```
(2) reqwest::async_impl::client: response '404 Not Found' for http://169.254.169.254/2020-10-27/meta-data/public-keys/1/openssh-key
```

I can update the error to say 404 for clarity, but I'm not quite sure if there is a simple way to pass the full uri. One option is to re-construct the uri via public constants, but I'm not a fan of that since it's not a guarantee that that was the uri that was `reqwest`'d

Reply:
I don't think the full URI is super helpful, since there's only one thing it could possibly be, given you know the version you're running.  And this case could seemingly only crop up if EC2 listed erroneous keys, or if the user changed the registered keys after the list was retrieved.  I don't think it's possible to change the keys associated in EC2 while an instance is running, so I don't expect that particular issue to come up.  The error seems fine to me...

Reply:
I think the error message `Error retrieving 'public-keys/1/openssh-key' from IMDS metadata` is clear enough. If you really do have a reproducible case where you need to know whether it was a 404 or something else, then you are really in the weeds anyway and it seems reasonable to expect the developer to turn on debugging for something that deep.